webview-ui/src/components/chat/ChatRow.tsx

```diff
@@ -600,10 +601,12 @@ export const ChatRowContent = ({
 									className="group"
 									onClick={() => vscode.postMessage({ type: "openFile", text: tool.content })}>
 									{tool.path?.startsWith(".") && <span>.</span>}
-									<span className="whitespace-nowrap overflow-hidden text-ellipsis text-left mr-2 rtl">
-										{removeLeadingNonAlphanumeric(tool.path ?? "") + "\u200E"}
-										{tool.reason}
-									</span>
+									<StandardTooltip content={`${tool.path}${tool.reason ? ` ${tool.reason}` : ""}`}>
```

Comment:
[P2] If tool.path is undefined the tooltip renders 'undefined'. Use a safe fallback and include an LRM to match the visible text’s directionality.

```suggestion
<StandardTooltip content={`${(tool.path ?? "")}${tool.reason ? ` ${tool.reason}` : ""}${(tool.path ?? "") || tool.reason ? "\u200E" : ""}`}>
```